src/JustEat.StatsD/Buffered/StatsDUtf8Formatter.cs

```diff
@@ -79,7 +99,15 @@ private static bool TryWritePayloadWithMessageKindSuffix(ref Buffer buffer, in S
                         // check if magnitude is integral, integers are written significantly faster
                         bool isMagnitudeIntegral = msg.Magnitude == integralMagnitude;
 
-                        var successSoFar = isMagnitudeIntegral ?
+                        bool successSoFar;
```

Comment:
Set this to `true` by default. Then each operation as I'd indicated below with the suggestions needs to `&=` so an early failure doesn't get overwritten by a later success.